Add tokens and currencies

herryho opened the pull request:
1. Add Tokens and Currencies modules to Bifrost runtime
2. Add LPToken type to Currency enum
3. Collect transaction fees into Treasury

herryho committed:
add orml tokens and currencies pallets to Bifrost Runtime

herryho committed:
add orml tokens and currencies pallets to Bifrost Runtime

herryho committed:
add Currency::LPToken and modify currency to index conversion method

herryho committed:
add Currency::LPToken and modify currency to index conversion method

herryho committed:
add Currency LPToken

herryho committed:
add Currency LPToken

herryho committed:
add call filter for Tokens and Currencies modules in Bifrost runtime

herryho committed:
add call filter for Tokens and Currencies modules in Bifrost runtime

yrong committed:
Fix build image[skip ci] (#210)

wodlj1314 committed:
update readme intro (#212)

* update readme intro

* fix the error

herryho committed:
fix bug in minter reward module (#214)

yooml committed:
docs: ✏️ Run full node with docker (#216)

herryho committed:
More benchmarking (#217)

* add flexible-fee benchmarking

* vtoken-mint benchmarking

* add flexible-fee benchmarking

* vtoken-mint benchmarking

* Minter-reward moudle: change auto depositing reward to user claiming reward

* minter-reward benchmarking part1

* finish minter reward benchmarking

* fix according to comments

ark930 committed:
Bifrost v0.8.1 (#220)

* disable on_runtime_upgrade in vesting

* bump version to 0.8.1

* Bump substrate (#213)

* Fix build image[skip ci] (#210)

* update readme intro (#212)

* update readme intro

* fix the error

* Bump substrate

Co-authored-by: wodlj1314 <72777624+wodlj1314@users.noreply.github.com>

* make MinVestedTransfer equal to ExistentialDeposit

* Bump version for cumulus

* fix types mismatch

* fix types mismatch

* fix vest start at

* fix

Co-authored-by: yrong1997 <yrong1997@gmail.com>
Co-authored-by: wodlj1314 <72777624+wodlj1314@users.noreply.github.com>

AllenPocketGamer commented on node/primitives/src/currency.rs:
```diff
@@ -120,14 +122,25 @@ macro_rules! create_currency_id {
+			// Currency 2 Discriminant:     1byte
+			// TokenSymbol 2 Discriminant:  1byte
+			// Currency 1 Discriminant:     1byte
+			// TokenSymbol 2 Discriminant:  1byte
```
tip: Currency 2 Discriminant

yrong committed:
Try-runtime integration (#221)

> Integate with command cli and runtime hooks
> Refact to use macro `with_runtime_or_err`
> Add entrance script
> Upate readme accordingly

herryho committed:
delete deref implementation for currencyid

herryho committed:
resolve conflicts

herryho committed:
format

herryho committed:
format
